File: examples/x1_rand_squares.py
#!/usr/bin/env python3
'''
@author  Michele Tomaiuolo - http://www.ce.unipr.it/people/tomamic
@license This software is free - http://www.gnu.org/licenses/gpl.html
'''

import g2d
from random import randrange

W, H = 640, 480
SIDE = 100
g2d.init_canvas((W, H))

n = int(g2d.prompt("n? "))
for i in range(n):
    r = randrange(255)
    g = randrange(255)
    b = randrange(255)
    x = randrange(W - SIDE)
    y = randrange(H - SIDE)
    g2d.set_color((r, g, b))
    g2d.fill_rect((x, y), (SIDE, SIDE))

g2d.main_loop()

File: go/x1_rand_squares.go
package main

import . "g2d"

func main() {
    w, h := 640, 480
    side := 100

    n := ToInt(Prompt("N? "))

    InitCanvas(Point{w, h})
    i := 0
    for i < n {
        r, g, b := RandInt(0, 255), RandInt(0, 255), RandInt(0, 255)
        x, y := RandInt(0, w-side), RandInt(0, h-side)
        SetColor(Color{r, g, b})
        FillRect(Point{x, y}, Point{side, side})
        i += 1
    }
    MainLoop()
}
